pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -421,6 +422,14 @@ func (c *Controller) injectPacket(tf *opsv1alpha1.Traceflow) error {
 		idICMP = uint16(tf.Spec.Packet.TransportHeader.ICMP.ID)
 		sequenceICMP = uint16(tf.Spec.Packet.TransportHeader.ICMP.Sequence)
 	}
+
+	var packetOutIcmpEchoRequestType icmpType
```

Comment:
Should we move this block to be inside: if tf.Spec.Packet.TransportHeader.ICMP != nil

Reply:
No, this type must be set when the protocol is ICMP/ICMPv6, not when the ICMP header is not nil.
And it is harmless if we force to set it always.